echidna/core/scale.py

```diff
@@ -58,25 +58,33 @@ def scale(self, spectrum, dimension, **kwargs):
         axis = spectrum.get_config().get_index(dimension)
         par = spectrum.get_config().get_par(dimension)
         low = par._low
+        high = par._high
         n_bins = par._bins
         step = par.get_width()
         for bin in range(n_bins):
             x = par.get_bin_centre(bin)
-            if x/sf < low or x/sf > par._high:
+            if x/sf < low or x/sf > high:
                 continue  # Trying to scale values outside range (Unknown)
+            old_bin1 = par.get_bin(x/sf)
+            old_bin_centre1 = par.get_bin_centre(old_bin1)
             y = interpolation(x/sf)
             if y <= 0.:
                 continue
-            old_bin1 = par.get_bin(x/sf)
-            old_bin_centre1 = par.get_bin_centre(old_bin1)
+            if numpy.isnan(y):
+                y = spectrum.project(dimension)[old_bin1]
             if par.get_bin_centre(old_bin1) > x/sf:
                 old_bin2 = old_bin1 - 1
                 if old_bin2 >= 0:
                     x_low1 = old_bin_centre1 - 0.5*step  # Equals x_high2
```

Comment:
I was wondering why this line uses `step` and not `step/sf`, since `old_bin_centre1` was found using `x/sf`. If you were to do:
```
x_low1 = old_bin_centre1 - 0.5*(step/sf)
```
and similar for line 80 below, this would include the events that are currently not included.